tools/privacyidea-user-action

```diff
@@ -0,0 +1,137 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from privacyidea.app import create_app
+import requests
+from privacyidea.lib.user import get_user_list
+from privacyidea.lib.token import get_tokens
+import jwt
+import click
+from privacyidea.lib.user import User
+
+
+__doc__ = """
+This is a script can be called from a cron job to enroll tokens to users,
+who do not have a token, yet.
+This could be a registration token or any other token type.
+The token is enrolled via the web UI, so that you can attach
+event handlers to the enrollment process.
+"""
+
+__version__ = "0.1"
+ROLE = "admin"
+ALGO = "RS256"
+
+app = create_app(config_name='production', silent=True)
+
+
+def create_jwt(jwt_privkey, jwt_user, jwt_realm):
+    """
+    create a JWT and return it
+    :param jwt_privkey:
+    :param jwt_user:
+    :param jwt_realm:
+    :return:
+    """
+    pkey = None
+    auth_token = None
+    with app.app_context():
+        if jwt_privkey:
+            with open(jwt_privkey, "r") as f:
+                pkey = f.read()
+            if pkey:
+                auth_token = jwt.encode(payload={"role": ROLE,
+                                     "username": jwt_user,
+                                     "realm": jwt_realm},
+                                     key=pkey,
+                                     algorithm=ALGO)
+    return auth_token
+
+
+@click.command()
+@click.option('-r', '--realm', required=True)
+@click.option('-t', '--tokentype', default="registration", show_default=True)
+@click.option('--jwt_privkey', required=True)
+@click.option('--jwt_user', required=True)
+@click.option('--jwt_realm')
+@click.option('--url', default="https://localhost", show_default=True)
+def enroll(realm, tokentype, jwt_privkey, jwt_user, jwt_realm, url):
+    """
+    Enroll a token for users in the given realm.
+
+    A token of type <tokentype> will be enrolled to all users
+    in the given realm, who currently have not token assigned.
+
+    The token is enrolled via the REST API, so events can be triggered.
+
+    """
+    auth_token = create_jwt(jwt_privkey, jwt_user, jwt_realm)
+    with app.app_context():
+        # Now we do all the users
+        users = get_user_list(param={"realm": realm})
```

Comment:
We should also document, that this lib call uses the corresponding resolver to query for the users. If the resolver has a limit on the amount of returned results, this will apply here as well.

Reply:
Will do

Reply:
done in 1909d9e9